src/combinators/box_body.rs

```diff
@@ -0,0 +1,61 @@
+use crate::Body;
+use bytes::Buf;
+use std::{
+    fmt,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// A boxed [`Body`] trait object.
+pub struct BoxBody<D, E> {
+    inner: Pin<Box<dyn Body<Data = D, Error = E> + Send + Sync + 'static>>,
```

Comment:
I wonder if we want to force this error to be `Box<dyn Error>` instead of a generic?

Reply:
How about making `E` default to `Box<dyn Error>` (i.e. `struct BoxBody<D, E = Box<dyn Error>>`) instead?

Reply:
Sounds like a good idea. Fixed in 7598a7699c537. I went for `Box<dyn Error + Send + Sync>` as that is what [tower uses](https://docs.rs/tower/0.4.5/tower/type.BoxError.html).